### PR TITLE
fix(LightAttachment): Stream can be disposed.

### PR DESCRIPTION
**Pull request by bruno-brant**
Should only be merged after #174.

### CODE REVIEW

**Comment by guilhermeluizsp** [comment]
@bruno-brant May you please rebase with the master branch?

**Comment by bruno-brant** [comment]
@guilhermeluizsp yeah, sorry, I was meaning to do it since last week. 

It's done now, and I'm correcting the other two issues as well.

File: src/Liquid.Base/Interfaces/Repository/ILightAttachment.cs [hunk 1]
```diff
@@ -1,17 +1,21 @@
...
+﻿// Copyright (c) Avanade Inc. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
 using System.IO;
 
 namespace Liquid.Interfaces
 {
-    public interface ILightAttachment
+    public interface ILightAttachment : IDisposable
```
Comment (guilhermeluizsp) [inline src/Liquid.Base/Interfaces/Repository/ILightAttachment.cs:9]: Just out of curiosity, isn't this change breaking the Interface Segregation Principle? Isn't being disposable an implementation detail (in this scenario)?

File: src/Liquid.Base/Interfaces/Telemetry/ILightTelemetry.cs [hunk 2]
```diff
@@ -10,7 +22,13 @@ public interface ILightTelemetry : IWorkbenchHealthCheck
...
         void EndMetricComputation(string metricLabel);
-        void EnqueueContext(string parentID, object value = null, string operationID = "");        
+        void EnqueueContext(string parentID, object value = null, string operationID = "");
         void DequeueContext();
+
+        /// <summary>
+        /// Tracks an exception from the application.
+        /// </summary>
+        /// <param name="e">The exception that will be tracked in the current context.</param>
+        void TrackException(Exception e);
```
Comment (guilhermeluizsp) [inline src/Liquid.Base/Interfaces/Telemetry/ILightTelemetry.cs]: Even though I agree that `ILightTelemetry` should have a `TrackException` method, I'm not sure if this change fits the context of this pull request. I'd say that you could undo these changes and create another PR just for that. Do you agree @bruno-brant?

File: src/Liquid.Base/Interfaces/Repository/ILightAttachment.cs (hunk 1 above)
Comment (bruno-brant) [reply src/Liquid.Base/Interfaces/Repository/ILightAttachment.cs:9]: I don't think so, because the interface is defining that the object owns a Stream, and the stream is disposable, so, by transition, ILightAttachment should be disposable. 

I would very much prefer not to have a Stream on this type - but this is the easy way out until we create a better interface for this.

Do you agree?

Comment (guilhermeluizsp) [reply src/Liquid.Base/Interfaces/Repository/ILightAttachment.cs:9]: You're right. The interface does have a reference to a `Stream` indeed.

File: src/Liquid.Base/Interfaces/Telemetry/ILightTelemetry.cs (hunk 2 above)
Comment (bruno-brant) [reply src/Liquid.Base/Interfaces/Telemetry/ILightTelemetry.cs]: That's from a previous change that I removed, sorry. I've removed the commit from this branch.

**Review by guilhermeluizsp — APPROVED** [review]
